src/components/QuranReader/EndOfSurahSection/StreakGoalCard/index.tsx

```diff
@@ -0,0 +1,144 @@
+import React from 'react';
```

Comment:
Cannot we reunse `src/components/HomePage/ReadingSection/StreakOrGoalCard/index.tsx`? I feel they are almost identical. 

We can extract the shared streak + goal layout into a reusable presentational component (e.g., `ReadingGoalCardContent`) that accepts props like `streak`, `goal`, `currentActivityDay`, `cta`, and interaction handlers. Let `StreakGoalCard` keep the data-fetching/mobile logic and render the shared component, while `StreakOrGoalCard` keeps the homepage-specific wrapper/link behavior but passes the same props. This removes the duplicated JSX and keeps context-specific logic separated.